client.go

```diff
@@ -61,3 +61,27 @@ func (c *Client) newRequest(method, requestPath string, body io.Reader) (*http.R
 	req.Header.Add("Content-Type", "application/json")
 	return req, err
 }
+
+func (c *Client) newQueryRequest(method, requestPath string, query string) (*http.Request, error) {
```

Comment:
Do we need this? Can query params be passed to `newRequest()` as part of the `requestPath` ?

Reply:
I tried that-- any special characters such as `?&=` for query parameters get URL-escaped if they're included in the `requestPath`.

Another option would to modify the existing `newRequest()` function and to support query parameters if that'd be better?

Reply:
I think it would be better to add to the existing method over creating a new one that is pretty much a copy of the other one. Or at least extract a common part.

If we're putting query params into the RawQuery, those params themselves need to be encoded, too. Is this something that the library is going to do?